kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt

```diff
@@ -0,0 +1,400 @@
+            // listen for fetch attachments to complete
+            attachmentsFuture.addDoneListener {
+                // get the feature attachments
+                val attachments = attachmentsFuture.get()
+                if (attachments.isNotEmpty()) {
+                    // create a new garden section data object and add it to the hash map of visited sections
+                    sectionsVisited[fenceFeatureName] =
+                        GardenSection(
+                            fenceFeature,
+                            fenceFeatureName,
+                            description,
+                            attachments
+                        )
+                    populateUI(
+                        sectionsVisited[fenceFeatureName],
+                        fenceGeotriggerNotificationInfo.geotriggerMonitor.geotrigger.name
+                    )
+                }
+            }
+            // if the section has been visited before, get it from the hash map
+        } else {
+            // this garden section has already been visited, show the information again
+            populateUI(
+                sectionsVisited[fenceFeatureName],
+                fenceGeotriggerNotificationInfo.geotriggerMonitor.geotrigger.name
+            )
+        }
+    }
+
+    /**
+     * Removes the [fenceFeatureName] from the garden section or
+     * the points of interest list using the corresponding [geotriggerType]
+     */
+    private fun removeFeatureInformation(fenceFeatureName: String, geotriggerType: String) {
+        if (geotriggerType == sectionGeotriggerName) {
+            sectionButton.text = "N/A"
+        } else {
+            poiList.remove(sectionsVisited[fenceFeatureName])
+            sectionsVisited[fenceFeatureName]?.title?.let { Log.e("REMOVED: ", it) }
+            poiListAdapter.notifyDataSetChanged()
+            if (poiList.size == 0) {
+                activityMainBinding.listAvailable.visibility = View.VISIBLE
+            }
+        }
+    }
+
+    /**
+     * Populates UI by setting [gardenSection] item based on the [geotriggerType]
+     */
+    private fun populateUI(
+        gardenSection: GardenSection?,
+        geotriggerType: String
+    ) {
+        if (gardenSection != null) {
+            if (geotriggerType == sectionGeotriggerName) {
+                sectionButton.text = gardenSection.title
+                sectionButton.setOnClickListener {
+                    if (gardenSection.title == sectionButton.text) {
+                        GardenDescriptionFragment(gardenSection, this).show(
+                            supportFragmentManager,
+                            "GardenDescriptionFragment"
+                        )
+                    }
+                }
+            } else {
+                poiList.add(gardenSection)
+                poiListAdapter.notifyDataSetChanged()
+                Log.e("ADDED: ", gardenSection.title)
```

Comment:
These should all go now that we've gotten to the bottom of the problem with missing POIs